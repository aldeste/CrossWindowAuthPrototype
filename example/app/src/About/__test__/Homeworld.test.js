import React from "react";
import renderer from "react-test-renderer";
import Homeworld from "../Homeworld";

jest.mock("../../Tags", () => ({
  Button: ({ className, children, ...props }) => (
    <button {...props}>{children}</button>
  ),
  Text: ({ className, children, ...props }) => <p {...props}>{children}</p>,
  Title: ({ className, children, ...props }) => <h1 {...props}>{children}</h1>,
  TitleH2: ({ className, children, ...props }) => (
    <h2 {...props}>{children}</h2>
  ),
  TitleH3: ({ className, children, ...props }) => (
    <h3 {...props}>{children}</h3>
  ),
  Form: ({ className, children, ...props }) => (
    <form {...props}>{children}</form>
  ),
  Label: ({ className, children, ...props }) => (
    <label {...props}>{children}</label>
  ),
  TextInput: ({ className, children, ...props }) => (
    <input {...props}>{children}</input>
  ),
  View: ({ className, children, ...props }) => <div {...props}>{children}</div>,
  Wrapper: ({ className, children, ...props }) => (
    <div {...props}>{children}</div>
  ),
  Iframe: ({ className, children, ...props }) => (
    <div {...props}>{children}</div>
  )
}));

const residents = [
  "Luke Skywalker",
  "C-3PO",
  "Darth Vader",
  "Owen Lars",
  "Beru Whitesun lars",
  "R5-D4",
  "Biggs Darklighter",
  "Anakin Skywalker",
  "Shmi Skywalker",
  "Cliegg Lars"
];

describe("About component", () => {
  it("Renders correctly", () => {
    expect(
      renderer
        .create(<Homeworld name={"Tatooine"} residents={residents} />)
        .toJSON()
    ).toMatchSnapshot();
  });
});
