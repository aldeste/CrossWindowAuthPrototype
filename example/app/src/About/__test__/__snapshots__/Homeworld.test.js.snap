// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component Renders correctly 1`] = `
<div>
  <p>
    You're from 
    Tatooine
    ! Other people are also from there! Maybe you know them?
  </p>
  <p>
    Luke Skywalker, C-3PO, Darth Vader, Owen Lars, Beru Whitesun lars, R5-D4, Biggs Darklighter, Anakin Skywalker, Shmi Skywalker, Cliegg Lars
  </p>
</div>
`;
